neo/rawio/edfrawio.py

```diff
@@ -174,6 +173,15 @@ def _parse_header(self):
         for array_key in array_keys:
             array_anno = {array_key: [h[array_key] for h in self.signal_headers]}
         seg_ann["signals"].append({"__array_annotations__": array_anno})
+        
+        # We store the following attributes for rapid access without needing the reader
+        
+        self._t_stop = self.edf_reader.datarecord_duration * self.edf_reader.datarecords_in_file
+        # use sample count of first signal in stream
+        self._stream_index_samples = {stream_index : self.edf_reader.getNSamples()[chidx][0] for stream_index, chidx in self.stream_idx_to_chidx.items()}
```

Comment:
Where is `[0]` coming from? 

this whole block seems like a great idea, but I don't the extra [0] in previous `_get_signal_size`?

Reply:
See https://github.com/NeuralEnsemble/python-neo/pull/1584/files#r1806665680